### PR TITLE
Update build/test

### PR DESCRIPTION
**Pull request by SuperQ**
* Use `make -C dir` instead of `cd dir && make`
* Cleanup uncessary working_directory in config.
* Update Go modules to latest.
* Drop support for old Go versions.

Signed-off-by: SuperQ <superq@gmail.com>

### CODE REVIEW

**Comment by SuperQ** [comment]
CC @mrueg

File: go.mod [hunk 1]
```diff
@@ -1,15 +1,15 @@
...
 	github.com/mwitkow/go-conntrack v0.0.0-20190716064945-2f068394615f
 	github.com/prometheus/client_golang v1.11.0
 	github.com/prometheus/client_model v0.2.0
-	golang.org/x/net v0.0.0-20210525063256-abc453219eb5
-	golang.org/x/oauth2 v0.0.0-20210514164344-f6687ab2804c
+	golang.org/x/net v0.0.0-20220105145211-5b0dc2dfae98
+	golang.org/x/oauth2 v0.0.0-20211104180415-d3ed0bb246c8
 	gopkg.in/alecthomas/kingpin.v2 v2.2.6
 	gopkg.in/yaml.v2 v2.4.0
 )
```
Comment (mrueg) [inline go.mod:15]: Do you also want to change the go 1.13 below to 1.15?

**Comment by mrueg** [comment]
the build tags in:
./config/http_config_test.go:// +build go1.8
./config/tls_config_test.go:// +build go1.8
./config/config_test.go:// +build go1.8
./config/http_config.go:// +build go1.8
and the logic in
https://github.com/prometheus/common/blob/main/Makefile#L19 could probably be cleaned up as well

File: go.mod (hunk 1 above)
Comment (SuperQ) [reply go.mod:15]: Yes, good point.

**Comment by SuperQ** [comment]
Ping @LeviHarrison @roidelapluie

File: go.mod [hunk 2]
```diff
@@ -1,17 +1,17 @@
...
 
-go 1.13
```
Comment (roidelapluie) [inline go.mod:17]: https://github.com/prometheus/client_golang/blob/main/go.mod#L15 Please update client_golang first

Comment (mrueg) [reply go.mod:17]: I made a PR against client_golang for that: https://github.com/prometheus/client_golang/pull/980

File: go.mod [hunk 3]
```diff
@@ -1,17 +1,17 @@
 module github.com/prometheus/common
 
 require (
-	github.com/go-kit/log v0.1.0
+	github.com/go-kit/log v0.2.0
 	github.com/golang/protobuf v1.4.3
 	github.com/julienschmidt/httprouter v1.3.0
 	github.com/matttproud/golang_protobuf_extensions v1.0.1
 	github.com/mwitkow/go-conntrack v0.0.0-20190716064945-2f068394615f
 	github.com/prometheus/client_golang v1.11.0
```
Comment (mrueg) [inline go.mod:9]: Can you update this one as well to v1.12.1 or later?

File: go.mod (hunk 2 above)
Comment (mrueg) [reply go.mod:17]: This one can be resolved, as the PR got merged.

File: go.mod (hunk 3 above)
Comment (SuperQ) [reply go.mod]: Done

**Comment by SuperQ** [comment]
Bleh, deprecated use linter errors.

**Comment by mrueg** [comment]
@SuperQ I assume those could be treated similar to
 ```"github.com/golang/protobuf/proto" //nolint:staticcheck // Ignore SA1019. Need to keep deprecated package for compatibility.```

Let me know if there's anything I can do to get this into a state in which it can be merged.

**Comment by SuperQ** [comment]
Ok, let me know if the changes to `expfmt` are acceptable.

**Review by roidelapluie — APPROVED** [review]
cc @beorn7 any opinion on the changes in expfmt?

**Comment by beorn7** [comment]
I think they are fine.

**Review by beorn7 — APPROVED** [review]

**Pull request merged**